crates/tabby/src/services/completion.rs

```diff
@@ -109,19 +109,38 @@ pub struct Segments {
     suffix: Option<String>,
 
     /// The relative path of the file that is being edited.
-    /// When git_url is set, this is the path of the file in the git repository.
-    /// When git_url is empty, this is the path of the file in the workspace.
+    /// - When `git_url` is set, this is the path of the file in the git repository.
+    /// - When `git_url` is empty, this is the path of the file in the workspace.
     filepath: Option<String>,
 
     /// The remote URL of the current git repository.
     /// Leave this empty if the file is not in a git repository,
     /// or the git repository does not have a remote URL.
     git_url: Option<String>,
 
+    /// The relevant declaration code snippets provided by editor.
+    /// The editor could invoke the language server to get relevant declarations.
+    declarations: Option<Vec<Declaration>>,
+
     /// Clipboard content when requesting code completion.
     clipboard: Option<String>,
 }
 
+/// A snippet of declaration code that is relevant to the current completion request.
+#[derive(Serialize, Deserialize, ToSchema, Clone, Debug)]
+pub struct Declaration {
+    /// Filepath of the file where the snippet is from.
+    /// This filepath is never the same as the current file (`segments.filepath`).
```

Comment:
```suggestion
```
It can be from same file as long as the declaration is not included in prefix / suffix.

Reply:
I think the prefix/suffix should be append to including the declaration in that case.